woocommerce/wc-admin
# Try: WooCommerce User Meta

This PR adds the ability to get user information, specifically meta, and update it from wc-admin.

I started looking into this as a part of #178, but it will be needed for persisting other user settings like https://github.com/woocommerce/wc-admin/issues/114.

There are more improvements that could be made here. For example, we could preload the user data on initial load with PHP. It looks like Gutenberg has some helpers we could use for this. There is also no error handling if the requests fail. I wasn't sure what we want to do in this case (or other critical requests) so I figured we could have a discussion about that and follow-up. I wanted to get a first-pass up in the meantime.

This PR specifically introduces an inbox last read time meta key. The integration there is basic since we are still deciding the future of the feature, but it works for saving and retrieving.

To Test:
* `npm install`
* `npm test` to make sure all tests pass
* `npm start`
* Open the inbox, verify the top notice is marked as unread.
* Switch to a different tab, then back to inbox. The orange dot should be gone.
* You can also monitor the Redux dev console, looking for `SET_USER`. You should see two requests. Make sure to select `wc-admin` in the instance selector (it defaults to core data). You can also verify the POST request worked via the Network console.
